source/php/Component/Hero/Hero.php

```diff
@@ -64,5 +64,28 @@ public function init()
         } else {
             $this->data['hasAnimation'] = false;
         }
+
+        /* Different hero types */
+        if($type) {
+            $this->data['contentSlotHasData'] = $this->slotHasData('content');
+            if(is_string($type)) {
+                $this->data['type'] = 'Hero.views.' . $type;
+                $this->data['classList'][] = $this->getBaseClass() . '--' . $type;
+            } else {
+                $this->data['type'] = 'Hero.views.' . $type['type'];
+                $this->data['classList'][] = $this->getBaseClass() . '--' . $type['type'];
+                $this->data['background'] = $type['background'] ? $type['background'] : ($type['image'] ? 'url(' .$type['image'] . ')' : '');
+                $this->data['imageSrc'] = $type['image'] ? $type['image'] : '';
+
+                if($type['modifiers']) {
```

Comment:
Split to another function. Do not fix a bad value (string) send a warning/notice about the issue.